### PR TITLE
Broadcast APIs - all of them

### PR DESCRIPTION
**Pull request by JayTokBox**

### CODE REVIEW

**Comment by msach22** [comment]
@JayTokBox looks like there are some merge conflicts, can you please update the PR?

File: lib/opentok/client.rb [hunk 1]
```diff
@@ -300,5 +300,96 @@ def layout_streams(session_id, opts)
...
+
+    def get_broadcast(broadcast_id)
+      response = self.class.get("/v2/project/#{@api_key}/broadcast/#{broadcast_id}", {
+          :headers => generate_headers
+      })
+      case response.code
+      when 200
+        response
+      when 400
+        raise OpenTokBroadcastError, "The broadcast could not be started. The request was invalid or invalid layout options or exceeded the limit of five simultaneous RTMP streams."
```
Comment (normanargueta) [inline lib/opentok/client.rb]: "The broadcast could not be started" I think is not the correct message for "get_broadcast" method

File: lib/opentok/client.rb [hunk 2]
```diff
@@ -300,5 +300,96 @@ def layout_streams(session_id, opts)
...
+
+    def stop_broadcast(broadcast_id)
+      response = self.class.post("/v2/project/#{@api_key}/broadcast/#{broadcast_id}/stop", {
+          :headers => generate_headers
+      })
+      case response.code
+      when 200
+        response
+      when 400
+        raise OpenTokBroadcastError, "The broadcast could not be started. The request was invalid or invalid layout options or exceeded the limit of five simultaneous RTMP streams."
```
Comment (normanargueta) [inline lib/opentok/client.rb]: "The broadcast could not be started" I think is not the correct message for "stop_broadcast" method

File: lib/opentok/client.rb [hunk 3]
```diff
@@ -300,5 +300,96 @@ def layout_streams(session_id, opts)
...
+      when 403
+        raise OpenTokAuthenticationError, "Authentication failed while starting a broadcast. API Key: #{@api_key}"
+      when 404
+        raise OpenTokBroadcastError, "No matching broadcast found (with the specified ID)"
+      when 500
+        raise OpenTokError, "OpenTok server error."
+      else
+        raise OpenTokBroadcastError, "The broadcast could not be started"
```
Comment (normanargueta) [inline lib/opentok/client.rb]: "The broadcast could not be started" I think is not the correct message for "get_broadcast" method

File: lib/opentok/client.rb [hunk 4]
```diff
@@ -300,5 +300,96 @@ def layout_streams(session_id, opts)
...
```
Comment (normanargueta) [inline lib/opentok/client.rb]: "The broadcast could not be started" I think is not the correct message for "stop_broadcast" method

File: lib/opentok/client.rb [hunk 5]
```diff
@@ -300,5 +300,96 @@ def layout_streams(session_id, opts)
...
+      opts.extend(HashExtensions)
+      response = self.class.put("/v2/project/#{@api_key}/broadcast/#{broadcast_id}/layout", {
+          :body => opts.camelize_keys!.to_json,
+          :headers => generate_headers("Content-Type" => "application/json")
+      })
+      case response.code
+      when 200
+        response
+      when 400
+        raise OpenTokBroadcastError, "The broadcast could not be started. The request was invalid or invalid layout options or exceeded the limit of five simultaneous RTMP streams."
```
Comment (normanargueta) [inline lib/opentok/client.rb]: "The broadcast could not be started" I think is not the correct message for "layout_broadcast" method

File: lib/opentok/client.rb [hunk 6]
```diff
@@ -300,5 +300,96 @@ def layout_streams(session_id, opts)
...
+      when 403
+        raise OpenTokAuthenticationError, "Authentication failed while starting a broadcast. API Key: #{@api_key}"
+      when 500
+        raise OpenTokError, "OpenTok server error."
+      else
+        raise OpenTokBroadcastError, "The broadcast could not be started"
```
Comment (normanargueta) [inline lib/opentok/client.rb]: "The broadcast could not be started" I think is not the correct message for "layout_broadcast" method

**Comment by normanargueta** [comment]
@JayTokBox please take a look at the inline comments, thanks!

**Pull request merged**